gateway/middleware.go

```diff
@@ -708,27 +707,28 @@ func (t BaseMiddleware) CheckSessionAndIdentityForValidKey(originalKey *string,
 		// update value of originalKey, as for custom-keys it might get updated (the key is generated again using alias)
 		*originalKey = key
 
+		session := session.Clone()
 		session.SetKeyHash(keyHash)
 		// If not in Session, and got it from AuthHandler, create a session with a new TTL
 		t.Logger().Info("Recreating session for key: ", obfuscateKey(key))
 
 		// cache it
-		clone := session.Clone()
+
 		if !t.Spec.GlobalConfig.LocalSessionCache.DisableCacheSessionState {
-			go SessionCache.Set(cacheKey, &clone, cache.DefaultExpiration)
+			SessionCache.Set(cacheKey, session, cache.DefaultExpiration)
```

Comment:
what will be the difference with goroutine and without?

Reply:
Previously, when it used pointer, it was racy. That's why initially removed it. Now without pointers `go` construct look safe, but I have not found any reasons to spawn goroutine there (and it is quite slow).